src/Reflection/Deprecation/Deprecation.php

```diff
@@ -0,0 +1,35 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Reflection\Deprecation;
+
+/**
+ * @api
+ */
+final class Deprecation
+{
+
+	private ?string $description = null;
+
+	private function __construct()
+	{
+	}
+
+	public static function create(): self
+	{
+		return new self();
+	}
+
+	public function getDescription(): ?string
+	{
+		return $this->description;
+	}
+
+	public function withDescription(?string $description): self
```

Comment:
* This should be static
* Called `createWithDescription`
* $description shouldn't be nullable